File: Code/linear_search.cpp
#include <bits/stdc++.h>
using namespace std;
#define vi vector<int>
#define vvi vector<vi>
#define pii pair<int, int>
#define vii vector<pii>
#define rep(i, a, b) for (int i = a; i < b; i++)
#define ff first
#define ss second
#define nl cout << endl;
const int N = 1e5 + 7, MOD = 1e9 + 7;

int main()
{
    int n;
    cout << "Enter size of array: ";
    cin >> n;
    int arr[n];
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
    }

    int key;
    cout << "Enter element to find: ";
    cin >> key;
    bool found = false;
    for (int i = 0; i < n; i++)
    {
        if (arr[i] == key)
        {
            printf("%d is found at index: %d", key, i);
            found = true;
        }
    }
    if (!found)
    {
        cout << "Element not found in the array!" << endl;
    }

    return 0;
}

File: linear_search.java
import java.util.Scanner;

public class linear_search {

  public static void main(String[] args) {
    int n;
    System.out.print("Enter size of array: ");
    Scanner scanner = new Scanner(System.in);
    n = scanner.nextInt();
    int[] arr = new int[n];
    for (int i = 0; i < n; i++) {
      System.out.print("Enter element at index " + i + ": ");
      arr[i] = scanner.nextInt();
    }

    int key;
    System.out.print("Enter element to find: ");
    key = scanner.nextInt();
    boolean found = false;
    for (int i = 0; i < n; i++) {
      if (arr[i] == key) {
        System.out.println(key + " is found at index: " + i);
        found = true;
      }
    }
    if (!found) {
      System.out.println("Element not found in the array!");
    }

    scanner.close();
  }
}
